src/panel/direction/MobileRoadMapPreview.jsx

```diff
@@ -1,63 +1,98 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import RoadMapStep from './RoadMapStep';
 import { fire } from 'src/libs/customEvents';
-import { CloseButton } from 'src/components/ui';
-
-export default class MobileRoadMapPreview extends React.Component {
-  static propTypes = {
-    steps: PropTypes.array.isRequired,
-    onClose: PropTypes.func.isRequired,
-  }
-
-  state = {
-    currentStep: 0,
-  }
-
-  componentDidMount() {
-    this.zoomToCurrentStep();
-    fire('move_mobile_geolocation_button', 65);
-  }
-
-  zoomToCurrentStep = () => {
-    fire('zoom_step', this.props.steps[this.state.currentStep]);
-  }
-
-  previous = () => {
-    this.setState(
-      prevState => ({ currentStep: Math.max(prevState.currentStep - 1, 0) }),
-      this.zoomToCurrentStep
-    );
-  }
+import classnames from 'classnames';
+import { Button } from '../../components/ui';
+
+const MobileRoadMapPreview = ({
+  steps,
+  onClose,
+}) => {
+  const [currentStep, setCurrentStep] = useState(0);
+  const [expanded, setExpanded] = useState(false);
+  const stepsRef = React.createRef();
+
+  const scroll = () => {
 
-  next = () => {
-    this.setState(
-      prevState => ({ currentStep: Math.min(prevState.currentStep + 1, this.props.steps.length) }),
-      this.zoomToCurrentStep
+    const newStep = Math.floor(
+      // Divide the step container's scrollLeft up to the middle of the screen with the size of a step
+      // to determine which step is present at the middle of the screen
+      (stepsRef.current.scrollLeft + window.innerWidth / 2) / (window.innerWidth - 70 + 12)
     );
-  }
-
-  render() {
-    const currentStep = this.props.steps[this.state.currentStep];
-
-    return <React.Fragment>
-      <div className="itinerary_mobile_step">
-        <CloseButton onClick={this.props.onClose} />
-        <RoadMapStep step={currentStep} />
-      </div>
-
-      <div className="itinerary_mobile_step_buttons">
-        <button
-          className="itinerary_mobile_step_button icon-chevron-left"
-          disabled={this.state.currentStep === 0}
-          onClick={this.previous}
-        />
-        <button
-          className="itinerary_mobile_step_button icon-chevron-right"
-          disabled={this.state.currentStep === this.props.steps.length - 1}
-          onClick={this.next}
-        />
-      </div>
-    </React.Fragment>;
-  }
-}
+
+    // If it has changed, save it and highlight it as the current step
+    if (currentStep !== newStep) {
+      setCurrentStep(newStep);
+      fire('zoom_step', steps[newStep]);
+    }
+  };
+
+  const toggleSize = () => {
+    setExpanded(!expanded);
+    fire('move_mobile_bottom_ui', stepsRef.current.offsetHeight);
+  };
+
+  // On load, zoom on step 0, unexpanded
+  useEffect(
```

Comment:
You don't need three separate `useEffect`. Two should be enough, because you have two side effects:
 - an event firing when `expanded` value change (including on first render)
 - an event firing when `currentStep` value change (including on first render)
  
```js
useEffect(
    () => { fire('move_mobile_bottom_ui', stepsRef.current.offsetHeight); },  // with probably a check for stepsRef because it will be null on first render
    [expanded /*, ... */]
);

useEffect(
    () => { fire('zoom_step', steps[currentStep]); },
    [currentStep /*, ... */]
);
```

In functional components you don't think in terms of mount=>updated=>unmount, but in terms of which component "state" (actually a mix of its React state, its props, its refs) value triggers which side effect. 
And don't worry, it's perfectly normal to go through these steps after time spent on class components :smiley: 
It took me sooo much time to understand most of the time it's simpler like this, and I'm still not fluent.